crates/valence_layer/src/message.rs

```diff
@@ -0,0 +1,276 @@
+use core::fmt;
+use std::convert::Infallible;
+use std::ops::Range;
+
+use valence_core::chunk_pos::{ChunkPos, ChunkView};
+
+use crate::bvh::{ChunkBvh, GetChunkPos};
+
+pub struct Messages<G, L> {
+    global: Vec<(G, Range<u32>)>,
+    local: Vec<(L, Range<u32>)>,
+    bvh: ChunkBvh<MessagePair<L>>,
+    staging: Vec<u8>,
+    ready: Vec<u8>,
+    is_ready: bool,
+}
+
+impl<G, L> Messages<G, L>
+where
+    G: Copy + Ord,
+    L: Copy + Ord + GetChunkPos,
+{
+    pub(crate) fn new() -> Self {
+        Self::default()
+    }
+
+    pub(crate) fn send_global<E>(
```

Comment:
This and `send_local` would be a good place to explain the difference between global and local messages